# Canadian Compensation Analysis

### Project Overview
This project analyzes Canadian wage and labour market data from Statistics Canada to explore compensation trends, salary benchmarks, and workforce patterns.

The analysis is designed to demonstrate how data can support compensation-related decisions through market research, data analysis, and evidence-based insights.

### Data Sources
This notebook uses two Statistics Canada datasets:

- **Table 14-10-0417-01:** Employee wages by occupation, annual
- **Table 14-10-0064-01:** Employee wages by industry, annual

### Objectives
The objectives of this notebook are to:

- Understand the structure and quality of the source data.
- Identify relevant variables for compensation analysis.
- Handle missing values, duplicates, and data quality issues.
- Prepare standardized datasets for further analysis.
- Create clean datasets for SQL analysis and Power BI visualization.

### Tools
- Python
- Pandas
- NumPy
- Jupyter Notebook

In [15]:
import pandas as pd
import numpy as np

occupation_path = "../data/raw/14100417.csv"
industry_path = "../data/raw/14100064.csv"

print("Paths created successfully.")

Paths created successfully.


#### Inspect Dataset Structure

To avoid loading the full datasets into memory, a small sample of rows is used to inspect column names, data types, and example values before selecting the variables needed for compensation analysis.

In [58]:
occupation_sample = pd.read_csv(occupation_path, nrows=10)
industry_sample = pd.read_csv(industry_path, nrows=10)

print("Occupation Dataset Shape (sample):")
print(occupation_sample.shape)

print("\nOccupation Dataset Columns:")
print(occupation_sample.columns.tolist())

print("\nIndustry Dataset Shape (sample):")
print(industry_sample.shape)

print("\nIndustry Dataset Columns:")
print(industry_sample.columns.tolist())

Occupation Dataset Shape (sample):
(10, 19)

Occupation Dataset Columns:
['REF_DATE', 'GEO', 'DGUID', 'Wages', 'Type of work', 'National Occupational Classification (NOC)', 'Gender', 'Age group', 'UOM', 'UOM_ID', 'SCALAR_FACTOR', 'SCALAR_ID', 'VECTOR', 'COORDINATE', 'VALUE', 'STATUS', 'SYMBOL', 'TERMINATED', 'DECIMALS']

Industry Dataset Shape (sample):
(10, 19)

Industry Dataset Columns:
['REF_DATE', 'GEO', 'DGUID', 'Wages', 'Type of work', 'North American Industry Classification System (NAICS)', 'Gender', 'Age group', 'UOM', 'UOM_ID', 'SCALAR_FACTOR', 'SCALAR_ID', 'VECTOR', 'COORDINATE', 'VALUE', 'STATUS', 'SYMBOL', 'TERMINATED', 'DECIMALS']


In [59]:
occupation_sample.head()

,REF_DATE,GEO,DGUID,Wages,Type of work,National Occupational Classification (NOC),Gender,Age group,UOM,UOM_ID,SCALAR_FACTOR,SCALAR_ID,VECTOR,COORDINATE,VALUE,STATUS,SYMBOL,TERMINATED,DECIMALS
0,1997,Canada,2021A000011124,"Total employees, all wages",Both full- and part-time employees,"Total employees, all occupations [00-95]",Total - Gender,15 years and over,Persons in thousands,428,units,0,v1481403365,1.1.1.1.1.1,11364.5,NaN,NaN,NaN,1
1,1997,Canada,2021A000011124,"Total employees, all wages",Both full- and part-time employees,"Total employees, all occupations [00-95]",Total - Gender,15 to 24 years,Persons in thousands,428,units,0,v1481403366,1.1.1.1.1.2,1877.8,NaN,NaN,NaN,1
2,1997,Canada,2021A000011124,"Total employees, all wages",Both full- and part-time employees,"Total employees, all occupations [00-95]",Total - Gender,25 to 54 years,Persons in thousands,428,units,0,v1481403367,1.1.1.1.1.3,8590.9,NaN,NaN,NaN,1
3,1997,Canada,2021A000011124,"Total employees, all wages",Both full- and part-time employees,"Total employees, all occupations [00-95]",Total - Gender,55 years and over,Persons in thousands,428,units,0,v1481403368,1.1.1.1.1.4,895.8,NaN,NaN,NaN,1
4,1997,Canada,2021A000011124,"Total employees, all wages",Both full- and part-time employees,"Total employees, all occupations [00-95]",Men+,15 years and over,Persons in thousands,428,units,0,v1481403369,1.1.1.1.2.1,5954.5,NaN,NaN,NaN,1


In [7]:
industry_sample.head()

,REF_DATE,GEO,DGUID,Wages,Type of work,North American Industry Classification System (NAICS),Gender,Age group,UOM,UOM_ID,SCALAR_FACTOR,SCALAR_ID,VECTOR,COORDINATE,VALUE,STATUS,SYMBOL,TERMINATED,DECIMALS
0,1997,Canada,2021A000011124,"Total employees, all wages",Both full- and part-time employees,"Total employees, all industries",Total - Gender,15 years and over,Persons in thousands,428,thousands,3,v2196614,1.1.1.1.1.1,11364.5,NaN,NaN,NaN,1
1,1997,Canada,2021A000011124,"Total employees, all wages",Both full- and part-time employees,"Total employees, all industries",Total - Gender,15 to 24 years,Persons in thousands,428,thousands,3,v2196899,1.1.1.1.1.2,1877.8,NaN,NaN,NaN,1
2,1997,Canada,2021A000011124,"Total employees, all wages",Both full- and part-time employees,"Total employees, all industries",Total - Gender,25 to 54 years,Persons in thousands,428,thousands,3,v2197184,1.1.1.1.1.3,8590.9,NaN,NaN,NaN,1
3,1997,Canada,2021A000011124,"Total employees, all wages",Both full- and part-time employees,"Total employees, all industries",Total - Gender,55 years and over,Persons in thousands,428,thousands,3,v2197469,1.1.1.1.1.4,895.8,NaN,NaN,NaN,1
4,1997,Canada,2021A000011124,"Total employees, all wages",Both full- and part-time employees,"Total employees, all industries",Men+,15 years and over,Persons in thousands,428,thousands,3,v2197754,1.1.1.1.2.1,5954.5,NaN,NaN,NaN,1


In [60]:
occupation_sample.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 10 entries, 0 to 9
Data columns (total 19 columns):
 #   Column                                      Non-Null Count  Dtype  
---  ------                                      --------------  -----  
 0   REF_DATE                                    10 non-null     int64  
 1   GEO                                         10 non-null     object 
 2   DGUID                                       10 non-null     object 
 3   Wages                                       10 non-null     object 
 4   Type of work                                10 non-null     object 
 5   National Occupational Classification (NOC)  10 non-null     object 
 6   Gender                                      10 non-null     object 
 7   Age group                                   10 non-null     object 
 8   UOM                                         10 non-null     object 
 9   UOM_ID                                      10 non-null     int64  
 10  SCALAR_FACTOR    

In [9]:
industry_sample.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 10 entries, 0 to 9
Data columns (total 19 columns):
 #   Column                                                 Non-Null Count  Dtype  
---  ------                                                 --------------  -----  
 0   REF_DATE                                               10 non-null     int64  
 1   GEO                                                    10 non-null     object 
 2   DGUID                                                  10 non-null     object 
 3   Wages                                                  10 non-null     object 
 4   Type of work                                           10 non-null     object 
 5   North American Industry Classification System (NAICS)  10 non-null     object 
 6   Gender                                                 10 non-null     object 
 7   Age group                                              10 non-null     object 
 8   UOM                                                  

#### Identify Key Variables and Categories

The Statistics Canada datasets contain several administrative and metadata fields that are not required for the compensation analysis.

The analysis focuses on variables related to time, geography, wage measures, employment type, occupation or industry, demographic characteristics, and reported wage values.

In [61]:
occupation_columns = [
    "REF_DATE",
    "GEO",
    "Wages",
    "Type of work",
    "National Occupational Classification (NOC)",
    "Gender",
    "Age group",
    "UOM",
    "VALUE"
]

industry_columns = [
    "REF_DATE",
    "GEO",
    "Wages",
    "Type of work",
    "North American Industry Classification System (NAICS)",
    "Gender",
    "Age group",
    "UOM",
    "VALUE"
]

#occupation dataset:
occupation_categories = {
    "REF_DATE": set(),
    "GEO": set(),
    "Wages": set(),
    "Type of work": set(),
    "Gender": set(),
    "Age group": set(),
    "UOM": set()
}

for chunk in pd.read_csv(
    occupation_path,
    usecols=occupation_columns,
    chunksize=200000
):
    for col in occupation_categories:
        occupation_categories[col].update(
            chunk[col].dropna().unique()
        )

for col, values in occupation_categories.items():
    print(f"\n{col}:")
    print(sorted(values))


REF_DATE:
[np.int64(1997), np.int64(1998), np.int64(1999), np.int64(2000), np.int64(2001), np.int64(2002), np.int64(2003), np.int64(2004), np.int64(2005), np.int64(2006), np.int64(2007), np.int64(2008), np.int64(2009), np.int64(2010), np.int64(2011), np.int64(2012), np.int64(2013), np.int64(2014), np.int64(2015), np.int64(2016), np.int64(2017), np.int64(2018), np.int64(2019), np.int64(2020), np.int64(2021), np.int64(2022), np.int64(2023), np.int64(2024), np.int64(2025)]

GEO:
['Alberta', 'British Columbia', 'Canada', 'Manitoba', 'New Brunswick', 'Newfoundland and Labrador', 'Nova Scotia', 'Ontario', 'Prince Edward Island', 'Quebec', 'Saskatchewan']

Wages:
['Average hourly gender wage ratio', 'Average hourly wage rate', 'Average weekly gender wage ratio', 'Average weekly wage rate', 'Median hourly gender wage ratio', 'Median hourly wage rate', 'Median weekly gender wage ratio', 'Median weekly wage rate', 'Total employees, all wages']

Type of work:
['Both full- and part-time employees

#### Define the Primary Compensation Benchmark

To ensure meaningful and consistent compensation comparisons, a standardized benchmark population was defined.

The primary benchmark focuses on:

- Full-time employees
- Total gender population
- Employees aged 15 years and over
- Median hourly wage rates
- Canadian provinces and the national benchmark
- 2015–2025 analysis period

Median hourly wages were selected as the primary compensation measure because the median is less sensitive to unusually high or low wage values and provides a useful representation of typical market compensation.

Additional datasets will be created separately for demographic and gender-based compensation analysis.

In [64]:
benchmark_chunks = []

for chunk in pd.read_csv(
    occupation_path,
    usecols=occupation_columns,
    chunksize=200000
):

    filtered = chunk[
        (chunk["REF_DATE"] >= 2015) &
        (chunk["Wages"] == "Median hourly wage rate") &
        (chunk["Type of work"] == "Full-time employees") &
        (chunk["Gender"] == "Total - Gender") &
        (chunk["Age group"] == "15 years and over") &
        (chunk["UOM"] == "Current dollars")
    ]

    benchmark_chunks.append(filtered)

occupation_benchmark = pd.concat(
    benchmark_chunks,
    ignore_index=True
)

occupation_benchmark.shape

(7139, 9)

In [65]:
occupation_benchmark.head()

,REF_DATE,GEO,Wages,Type of work,National Occupational Classification (NOC),Gender,Age group,UOM,VALUE
0,2015,Canada,Median hourly wage rate,Full-time employees,"Total employees, all occupations [00-95]",Total - Gender,15 years and over,Current dollars,24.04
1,2015,Canada,Median hourly wage rate,Full-time employees,"Management occupations [00, 10, 20, 30, 40, 50...",Total - Gender,15 years and over,Current dollars,41.37
2,2015,Canada,Median hourly wage rate,Full-time employees,Legislative and senior management occupations [0],Total - Gender,15 years and over,Current dollars,57.69
3,2015,Canada,Median hourly wage rate,Full-time employees,"Specialized middle management occupations [10,...",Total - Gender,15 years and over,Current dollars,45.67
4,2015,Canada,Median hourly wage rate,Full-time employees,Middle management occupations in retail and wh...,Total - Gender,15 years and over,Current dollars,28.85


In [66]:
occupation_benchmark.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 7139 entries, 0 to 7138
Data columns (total 9 columns):
 #   Column                                      Non-Null Count  Dtype  
---  ------                                      --------------  -----  
 0   REF_DATE                                    7139 non-null   int64  
 1   GEO                                         7139 non-null   object 
 2   Wages                                       7139 non-null   object 
 3   Type of work                                7139 non-null   object 
 4   National Occupational Classification (NOC)  7139 non-null   object 
 5   Gender                                      7139 non-null   object 
 6   Age group                                   7139 non-null   object 
 7   UOM                                         7139 non-null   object 
 8   VALUE                                       7017 non-null   float64
dtypes: float64(1), int64(1), object(7)
memory usage: 502.1+ KB


In [67]:
occupation_benchmark["REF_DATE"].value_counts().sort_index()

REF_DATE
2015    649
2016    649
2017    649
2018    649
2019    649
2020    649
2021    649
2022    649
2023    649
2024    649
2025    649
Name: count, dtype: int64

In [20]:
occupation_benchmark["GEO"].value_counts()

GEO
Canada                       649
Newfoundland and Labrador    649
Prince Edward Island         649
Nova Scotia                  649
New Brunswick                649
Quebec                       649
Ontario                      649
Manitoba                     649
Saskatchewan                 649
Alberta                      649
British Columbia             649
Name: count, dtype: int64

In [69]:
#number of unique NOC values
occupation_benchmark[
    "National Occupational Classification (NOC)"
].nunique()

59

In [70]:
#look at some of them (the first ~30 unique NOC values)
occupation_benchmark[
    "National Occupational Classification (NOC)"
].unique()[:30]

array(['Total employees, all occupations [00-95]',
       'Management occupations [00, 10, 20, 30, 40, 50, 60, 70, 80, 90]',
       'Legislative and senior management occupations [0]',
       'Specialized middle management occupations [10, 20, 30, 40, 50]',
       'Middle management occupations in retail and wholesale trade and customer services [60]',
       'Middle management occupations in trades, transportation, production and utilities [70, 80, 90]',
       'Business, finance and administration occupations, except management [11-14]',
       'Professional occupations in finance and business [11]',
       'Professional occupations in finance [111]',
       'Professional occupations in business [112]',
       'Administrative and financial supervisors and specialized administrative occupations [12]',
       'Administrative occupations and transportation logistics occupations [13]',
       'Administrative and financial support and supply chain logistics occupations [14]',
       'Natu

#### Standardize Occupational Classification Level

The occupation dataset contains multiple levels of the National Occupational Classification (NOC) hierarchy, ranging from broad occupational categories to more detailed occupational groups.

Comparing wage levels across different hierarchy levels could produce misleading results. Therefore, occupational codes are extracted from the occupation descriptions so that comparable classification levels can be identified and used for compensation benchmarking.

In [71]:
#Extract the NOC code
occupation_benchmark["NOC_Code"] = (
    occupation_benchmark["National Occupational Classification (NOC)"]
    .str.extract(r"\[([^\]]+)\]")
)

occupation_benchmark[
    ["National Occupational Classification (NOC)", "NOC_Code"]
].head(20)

,National Occupational Classification (NOC),NOC_Code
0,"Total employees, all occupations [00-95]",00-95
1,"Management occupations [00, 10, 20, 30, 40, 50...","00, 10, 20, 30, 40, 50, 60, 70, 80, 90"
2,Legislative and senior management occupations [0],0
3,"Specialized middle management occupations [10,...","10, 20, 30, 40, 50"
4,Middle management occupations in retail and wh...,60
5,"Middle management occupations in trades, trans...","70, 80, 90"
6,"Business, finance and administration occupatio...",11-14
7,Professional occupations in finance and busine...,11
8,Professional occupations in finance [111],111
9,Professional occupations in business [112],112


In [72]:
#Determine the code structure
occupation_benchmark["NOC_Code"].value_counts().sort_index()

NOC_Code
0                                         121
00, 10, 20, 30, 40, 50, 60, 70, 80, 90    121
00-95                                     121
10, 20, 30, 40, 50                        121
11                                        121
11-14                                     121
111                                       121
112                                       121
12                                        121
13                                        121
14                                        121
21                                        121
21-22                                     121
211                                       121
212                                       121
213                                       121
22                                        121
31                                        121
31-33                                     121
311                                       121
312                                       121
313                      

In [73]:
#rather than manually looking through everything, let's classify the codes.
def classify_noc(code):
    if pd.isna(code):
        return "Unknown"
    elif code.isdigit():
        return f"{len(code)}-digit"
    else:
        return "Aggregate/Multiple"

occupation_benchmark["NOC_Level"] = (
    occupation_benchmark["NOC_Code"].apply(classify_noc)
)

occupation_benchmark["NOC_Level"].value_counts()

NOC_Level
2-digit               3146
Aggregate/Multiple    2178
3-digit               1694
1-digit                121
Name: count, dtype: int64

In [74]:
#check how many occupational categories exist at each level
occupation_benchmark.groupby("NOC_Level")[
    "National Occupational Classification (NOC)"
].nunique()

NOC_Level
1-digit                1
2-digit               26
3-digit               14
Aggregate/Multiple    18
Name: National Occupational Classification (NOC), dtype: int64

In [75]:
#inspect the 3-digit occupations
three_digit_occupations = (
    occupation_benchmark.loc[
        occupation_benchmark["NOC_Level"] == "3-digit",
        ["NOC_Code", "National Occupational Classification (NOC)"]
    ]
    .drop_duplicates()
    .sort_values("NOC_Code")
)

three_digit_occupations

,NOC_Code,National Occupational Classification (NOC)
8,111,Professional occupations in finance [111]
9,112,Professional occupations in business [112]
15,211,Professional occupations in natural sciences [...
16,212,Professional occupations in applied sciences (...
17,213,Professional occupations in engineering [213]
21,311,Health treating and consultation services prof...
22,312,Therapy and assessment professionals [312]
23,313,Nursing and allied health professionals [313]
28,411,Professional occupations in law [411]
29,412,Professional occupations in education services...


The cleaned benchmark currently has:
7,139 observations
59 occupational classifications
14 three-digit occupational groups
2015–2025
11 geographic levels (Canada + 10 provinces)

The repeated count of 121 per NOC code also makes sense: 11 years × 11 geographies = 121 observations.

We'll now keep only the 14 three-digit NOC groups for our main occupational comparison.

#### Create a Comparable Occupational Benchmark

The source data contains multiple levels of the National Occupational Classification (NOC) hierarchy. To improve comparability, the primary occupational benchmark is restricted to three-digit NOC groups.

This produces a consistent set of occupational groups that can be compared across provinces and over time without mixing broad occupational categories with more detailed classifications.

The resulting benchmark will be used to evaluate:

- Compensation levels across occupational groups
- Provincial differences in compensation
- Nova Scotia wages relative to the Canadian benchmark
- Long-term wage trends
- Market competitiveness

In [76]:
#create the dataset:
occupation_3digit = occupation_benchmark[
    occupation_benchmark["NOC_Level"] == "3-digit"
].copy()

occupation_3digit.shape

(1694, 11)

In [77]:
#reate a new column:
occupation_3digit["Occupation"] = (
    occupation_3digit["National Occupational Classification (NOC)"]
    .str.replace(r"\s*\[[^\]]+\]$", "", regex=True)
    .str.strip()
)

In [78]:
occupation_3digit[
    ["NOC_Code", "Occupation"]
].drop_duplicates().sort_values("NOC_Code")

,NOC_Code,Occupation
8,111,Professional occupations in finance
9,112,Professional occupations in business
15,211,Professional occupations in natural sciences
16,212,Professional occupations in applied sciences (...
17,213,Professional occupations in engineering
21,311,Health treating and consultation services prof...
22,312,Therapy and assessment professionals
23,313,Nursing and allied health professionals
28,411,Professional occupations in law
29,412,Professional occupations in education services


## Standardize Column Names

Column names are standardized to improve readability and maintain consistency across Python, SQL, and Power BI.

In [79]:
occupation_3digit = occupation_3digit.rename(columns={
    "REF_DATE": "year",
    "GEO": "geography",
    "Wages": "wage_measure",
    "Type of work": "work_type",
    "Gender": "gender",
    "Age group": "age_group",
    "UOM": "unit",
    "VALUE": "median_hourly_wage",
    "NOC_Code": "noc_code",
    "NOC_Level": "noc_level"
})

In [80]:
#Now we no longer need the original long NOC column:
occupation_3digit = occupation_3digit.drop(
    columns=["National Occupational Classification (NOC)"]
)

In [81]:
occupation_3digit.head()

,year,geography,wage_measure,work_type,gender,age_group,unit,median_hourly_wage,noc_code,noc_level,Occupation
8,2015,Canada,Median hourly wage rate,Full-time employees,Total - Gender,15 years and over,Current dollars,32.19,111,3-digit,Professional occupations in finance
9,2015,Canada,Median hourly wage rate,Full-time employees,Total - Gender,15 years and over,Current dollars,32.69,112,3-digit,Professional occupations in business
15,2015,Canada,Median hourly wage rate,Full-time employees,Total - Gender,15 years and over,Current dollars,35.90,211,3-digit,Professional occupations in natural sciences
16,2015,Canada,Median hourly wage rate,Full-time employees,Total - Gender,15 years and over,Current dollars,36.41,212,3-digit,Professional occupations in applied sciences (...
17,2015,Canada,Median hourly wage rate,Full-time employees,Total - Gender,15 years and over,Current dollars,41.21,213,3-digit,Professional occupations in engineering


In [82]:
occupation_3digit.info()

<class 'pandas.core.frame.DataFrame'>
Index: 1694 entries, 8 to 7114
Data columns (total 11 columns):
 #   Column              Non-Null Count  Dtype  
---  ------              --------------  -----  
 0   year                1694 non-null   int64  
 1   geography           1694 non-null   object 
 2   wage_measure        1694 non-null   object 
 3   work_type           1694 non-null   object 
 4   gender              1694 non-null   object 
 5   age_group           1694 non-null   object 
 6   unit                1694 non-null   object 
 7   median_hourly_wage  1677 non-null   float64
 8   noc_code            1694 non-null   object 
 9   noc_level           1694 non-null   object 
 10  Occupation          1694 non-null   object 
dtypes: float64(1), int64(1), object(9)
memory usage: 158.8+ KB


#### Data Quality Assessment

The standardized benchmark dataset is evaluated for missing values, duplicate records, invalid wage values, and consistency across years and geographic regions.

In [83]:
#Missing values
occupation_3digit.isnull().sum()

year                   0
geography              0
wage_measure           0
work_type              0
gender                 0
age_group              0
unit                   0
median_hourly_wage    17
noc_code               0
noc_level              0
Occupation             0
dtype: int64

Only 17 of 1,694 observations have missing wage values—about 1%—so the dataset is quite complete.

In [84]:
#Duplicates
occupation_3digit.duplicated().sum()

np.int64(0)

In [85]:
#Wage statistics
occupation_3digit["median_hourly_wage"].describe()

count    1677.000000
mean       39.346231
std         8.625620
min        15.000000
25%        34.970000
50%        39.560000
75%        43.800000
max        96.150000
Name: median_hourly_wage, dtype: float64

In [86]:
#Check years
occupation_3digit["year"].value_counts().sort_index()

year
2015    154
2016    154
2017    154
2018    154
2019    154
2020    154
2021    154
2022    154
2023    154
2024    154
2025    154
Name: count, dtype: int64

In [87]:
#Check geography
occupation_3digit["geography"].value_counts()

geography
Canada                       154
Newfoundland and Labrador    154
Prince Edward Island         154
Nova Scotia                  154
New Brunswick                154
Quebec                       154
Ontario                      154
Manitoba                     154
Saskatchewan                 154
Alberta                      154
British Columbia             154
Name: count, dtype: int64

In [88]:
#check missing wage observations
missing_wages = occupation_3digit[
    occupation_3digit["median_hourly_wage"].isna()
]

missing_wages[
    ["year", "geography", "noc_code", "Occupation"]
].head(20)

,year,geography,noc_code,Occupation
140,2015,Prince Edward Island,312,Therapy and assessment professionals
146,2015,Prince Edward Island,411,Professional occupations in law
151,2015,Prince Edward Island,421,Occupations in front-line public protection se...
736,2016,Newfoundland and Labrador,411,Professional occupations in law
789,2016,Prince Edward Island,312,Therapy and assessment professionals
795,2016,Prince Edward Island,411,Professional occupations in law
1385,2017,Newfoundland and Labrador,411,Professional occupations in law
1438,2017,Prince Edward Island,312,Therapy and assessment professionals
1444,2017,Prince Edward Island,411,Professional occupations in law
2028,2018,Newfoundland and Labrador,312,Therapy and assessment professionals


In [89]:
occupation_3digit.groupby("geography")[
    "median_hourly_wage"
].apply(lambda x: x.isna().sum()).sort_values(ascending=False)

geography
Prince Edward Island         14
Newfoundland and Labrador     3
Alberta                       0
Canada                        0
British Columbia              0
New Brunswick                 0
Manitoba                      0
Nova Scotia                   0
Ontario                       0
Quebec                        0
Saskatchewan                  0
Name: median_hourly_wage, dtype: int64

#### Missing Value Analysis

The benchmark dataset contains a small number of missing wage observations. Before determining how to handle these records, their distribution across geography, year, and occupational group is examined.

Missing Statistics Canada observations are not automatically imputed because unavailable or suppressed estimates may reflect insufficient sample sizes or data-quality considerations.

In [90]:
missing_wages = occupation_3digit[
    occupation_3digit["median_hourly_wage"].isna()
]

missing_wages[
    ["year", "geography", "noc_code", "Occupation", "median_hourly_wage"]
]

,year,geography,noc_code,Occupation,median_hourly_wage
140,2015,Prince Edward Island,312,Therapy and assessment professionals,NaN
146,2015,Prince Edward Island,411,Professional occupations in law,NaN
151,2015,Prince Edward Island,421,Occupations in front-line public protection se...,NaN
736,2016,Newfoundland and Labrador,411,Professional occupations in law,NaN
789,2016,Prince Edward Island,312,Therapy and assessment professionals,NaN
795,2016,Prince Edward Island,411,Professional occupations in law,NaN
1385,2017,Newfoundland and Labrador,411,Professional occupations in law,NaN
1438,2017,Prince Edward Island,312,Therapy and assessment professionals,NaN
1444,2017,Prince Edward Island,411,Professional occupations in law,NaN
2028,2018,Newfoundland and Labrador,312,Therapy and assessment professionals,NaN


In [91]:
missing_wages["geography"].value_counts()

geography
Prince Edward Island         14
Newfoundland and Labrador     3
Name: count, dtype: int64

In [92]:
missing_wages["year"].value_counts().sort_index()

year
2015    3
2016    3
2017    3
2018    2
2019    1
2020    2
2021    1
2022    1
2025    1
Name: count, dtype: int64

In [93]:
#Validate uniqueness
duplicate_keys = occupation_3digit.duplicated(
    subset=["year", "geography", "noc_code"],
    keep=False
)

occupation_3digit[duplicate_keys].shape

(0, 11)

In [94]:
occupation_3digit.duplicated().sum()

np.int64(0)

#### Export Cleaned Analytical Datasets

The cleaned occupational benchmark and compensation scorecard are exported for use in subsequent SQL analysis and Power BI dashboard development.

In [95]:
occupation_3digit.to_csv(
    "../data/processed/occupation_benchmark_clean.csv",
    index=False
)
